File: test-suite/src/test/java/io/micronaut/kafka/docs/producer/config/ClientIdClient.java
package io.micronaut.kafka.docs.producer.config;

import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "ClientIdClientTest")
// tag::annotation[]
@KafkaClient("product-client")
// end::annotation[]
public interface ClientIdClient {
    // define client API
}

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/config/ClientIdClient.py
from abc import ABC

from micronaut.configuration.kafka.annotation import KafkaClient
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="ClientIdClientTest")
# tag::annotation[]
@KafkaClient("product-client")
# end::annotation[]
class ClientIdClient(ABC):
    # define client API
    pass
